src/internal.c

```diff
@@ -8635,10 +8635,26 @@ static int BuildUserAuthRequestEcc(WOLFSSH* ssh,
     word32 begin;
     enum wc_HashType hashId = WC_HASH_TYPE_SHA;
     int ret = WS_SUCCESS;
-    byte* r;
-    byte* s;
-    byte sig[139]; /* wc_ecc_sig_size() for a prime521 key. */
-    word32 sigSz = sizeof(sig), rSz, sSz;
+    byte* r_ptr;
+    byte* s_ptr;
+    byte* sig_ptr;
+    word32 rSz = ECC_MAX_SIG_SIZE / 2;
+    word32 sSz = ECC_MAX_SIG_SIZE / 2;
+    word32 sigSz = ECC_MAX_SIG_SIZE;
+#ifdef WOLFSSH_SMALL_STACK
+    r_ptr = (byte*)WMALLOC(rSz, heap, DYNTYPE_BUFFER);
```

Comment:
Jenkins tests are having an issue with 'heap', looks like it needs to come from the 'ssh' struct if being used.